src/index.ts

```diff
@@ -673,8 +685,182 @@ export interface DocumentContent {
+
+export type Operator = '+' | '-' | '*' | '/';
+
+export type TemplatePositionValue<
+  Constants extends BaseConstants = BaseConstants,
+  V extends TemplatePositionBaseValue<Constants> = TemplatePositionBaseValue<Constants>
+> =
+  | V
+  | `${V} ${Operator} ${V}`
+  | `${V} ${Operator} ${V} ${Operator} ${V}`
+  | `${V} ${Operator} ${V} ${Operator} ${V} ${Operator} ${V}`;
+
+/** A text node, similar to SVG <text> */
+export interface TemplateTextItem<Constants extends BaseConstants> extends TemplateItemCommon {
+  /** text node */
+  type: 'text';
+  /** the text to render */
+  text: string;
+  /** optional font size in pixels */
+  fontSize?: number;
+  /** optional stroke width in pixels */
+  strokeWidth?: number;
+  /** should the text be bolded */
+  bold?: boolean;
+  /** the position at which to draw (aligned to top+left) */
+  position: {
+    /** the x coordinate at which to begin drawing */
+    x: TemplatePositionValue<Constants>;
+    /** the y coordinate at which to begin drawing */
+    y: TemplatePositionValue<Constants>;
+  };
+}
+
+/** Represents a single template drawing/implementation item */
+export type TemplateItem<Constants extends BaseConstants = BaseConstants> = TemplateGroupItem<Constants> | TemplatePathItem<Constants> | TemplateTextItem<Constants>;
+
+const templateItem = discriminator('type', {
+    group: properties({
+      id: string(),
+      boundingBox: properties({
+        x: empty() as CompiledSchema<TemplatePositionValue, unknown>,
+        y: empty() as CompiledSchema<TemplatePositionValue, unknown>,
+        width: empty() as CompiledSchema<TemplatePositionValue, unknown>,
+        height: empty() as CompiledSchema<TemplatePositionValue, unknown>,
+      }),
+      /** TS JTD cannot express cyclical definitions */
+      children: elements(empty() as CompiledSchema<TemplateItem, unknown>),
+    }, {
+      repeat: properties(undefined, {
+        rows: uint8(),
+        cols: uint8(),
+      }),
+    }),
+    path: properties({
+      id: string(),
+      data: elements(empty() as CompiledSchema<TemplateItemPathEntry, unknown>),
+    }, {
+      strokeWidth: uint8(),
+    }),
+    text: properties({
+      text: string(),
+      position: properties({
+        x: empty() as CompiledSchema<TemplatePositionValue, unknown>,
+        y: empty() as CompiledSchema<TemplatePositionValue, unknown>,
+      }),
+    }, {
+      fontSize: uint8(),
+      strokeWidth: uint8(),
+      bold: boolean(),
+    })
+}) satisfies CompiledSchema<TemplateItem, unknown>;
+
+const templateContent = properties({
+  name: string(),
+  author: string(),
+  iconData: string(),
+  categories: elements(string()),
+  labels: elements(string()),
+  orientation: enumeration("portrait", "landscape"),
+  templateVersion: string(),
+  formatVersion: uint8(),
+  supportedScreens: elements(enumeration('rm2', 'rmPP')),
+  constants: elements(values(int32())),
+  items: elements(templateItem)
```

Comment:
if you make the above change, this can be simplified to

```suggestion
  items: elements(empty())
```

and almost everything else can be deleted.